helm/cert-manager-app/templates/controller-deployment.yaml

```diff
@@ -47,6 +47,10 @@ spec:
           - --default-issuer-kind={{ .Values.controller.defaultIssuer.kind }}
           - --default-issuer-group={{ .Values.controller.defaultIssuer.group }}
           {{- end }}
+          {{- if eq .Values.Installation.V1.GiantSwarm.CertManager.AcmeSolver "dns01" }}
+          - --dns01-recursive-nameservers-only
+          - --dns01-recursive-nameservers="{{ .Values.Installation.V1.GiantSwarm.CertManager.AcmeSolver.DNSServer }}:53"
```

Comment:
Should we check if that value exists in the `if` above?

Reply:
Feels like they should always be defined together so I think it's fine without a check.

Reply:
But what happens if someone accidentally forgets to set `DNSServer` or has a typo or indentation issue? :) I mean should we fail early to prevent a potentially longer debugging session or is the fallback behavior good enough and maybe easily to spot in the log output? I don't have a strong opinion here. It's more like as a default, always check for all values you use. On the other hand, that can make Helm templates look overloaded and hard to read. So fine with whatever you decide on.

Reply:
It will be defined as part of the installation setup, so no need to check it here as it'll never need to be configured on an app-level.

Reply:
I see where you're coming from. If we set `AcmeSolver` to "dns01" and leave `DNSServer` empty, then we'll have final flags of `--dns01-recursive-nameservers-only  --dns01-recursive-nameservers=""` so DNS checks would likely fail. We'd see this failure in the logs, but it might not be clear what the problem is. Maybe there's a prometheus metric that would indicate a problem? Eventually certs wouldn't be issued and we'd get alerts when services go down. The better solution in my view would be to have validation checks on `installations` settings to catch mutually exclusive values like this, but I can't see us implementing that any time soon.

Reply:
If the IP isn't configured in installations then cert manager will just fail to start with invalid args - it should be pretty obvious what's going on. Agreed that error checking here isn't really the right place.

Reply:
Most elegant in the end would be to dive deeper into json-schema and use [property dependencies](http://json-schema.org/understanding-json-schema/reference/object.html#property-dependencies) for `DNSServer` or [enumerated values](http://json-schema.org/understanding-json-schema/reference/generic.html#enumerated-values) for `AcmeSolver`. But a more complex topic and better to be handled upstream..